Tests/MathLib/TestPiecewiseLinearCurve.cpp

```diff
@@ -0,0 +1,125 @@
+/**
+ * \copyright
+ * Copyright (c) 2012-2016, OpenGeoSys Community (http://www.opengeosys.org)
+ *            Distributed under a Modified BSD License.
+ *              See accompanying file LICENSE.txt or
+ *              http://www.opengeosys.org/project/license
+ *
+ * \file   TestPiecewiseLinearCurve.cpp
+ *
+ * Created on November 11, 2016, 10:58 AM
+ */
+
+#include <cmath>
+#include <limits>
+
+#include "gtest/gtest.h"
+#include "TestTools.h"
+
+#include "BaseLib/ConfigTree.h"
+#include "MathLib/Curve/PiecewiseLinearCurve.h"
+#include "MathLib/Curve/CreatePiecewiseLinearCurve.h"
+
+std::unique_ptr<MathLib::PiecewiseLinearCurve> createPiecewiseLinearCurve(
+    const char xml[])
+{
+    auto const ptree = readXml(xml);
+    BaseLib::ConfigTree conf(ptree, "", BaseLib::ConfigTree::onerror,
+                             BaseLib::ConfigTree::onwarning);
+    auto const& sub_config = conf.getConfigSubtree("curve");
+    return MathLib::createPiecewiseLinearCurve(sub_config);
+}
+
+TEST(MathLibCurve, PiecewiseLinearCurveParsing)
+{
+    const char xml[] =
+        "<curve>"
+        "   <type>PiecewiseLinear</type>"
+        "   <point><data> 0.2 20.</data></point> "
+        "   <point><data> 0.4 10.</data></point> "
+        "   <point><data> 0.5 5.</data></point> "
+        "   <point><data> 0.6 3.</data></point> "
+        "   <point><data> 0.7 2.</data></point> "
+        "</curve>";
```

Comment:
This is not the curve format that is already implemented. Do you plan to change this?